source/extensions/filters/network/mongo_proxy/proxy.cc

```diff
@@ -366,11 +366,11 @@ void ProxyFilter::onEvent(Network::ConnectionEvent event) {
   }
 
   if (event == Network::ConnectionEvent::RemoteClose && !active_query_list_.empty()) {
-    stats_.cx_destroy_local_with_active_rq_.inc();
+    stats_.cx_destroy_remote_with_active_rq_.inc();
   }
 
   if (event == Network::ConnectionEvent::LocalClose && !active_query_list_.empty()) {
-    stats_.cx_destroy_remote_with_active_rq_.inc();
+    stats_.cx_destroy_local_with_active_rq_.inc();
```

Comment:
This change triggered some continuous integration test failures in test/extensions/filters/network/mongo_proxy/proxy_test.cc due to tests that covered the swapped behavior.  I think the solution is to also swap the counter name in the test expectations.

2020-11-12T06:48:59.5799388Z [ RUN      ] MongoProxyFilterTest.ConnectionDestroyLocal
2020-11-12T06:48:59.5800234Z TestRandomGenerator running with seed -291122280
2020-11-12T06:48:59.5800928Z test/extensions/filters/network/mongo_proxy/proxy_test.cc:629: Failure
2020-11-12T06:48:59.5801543Z Expected equality of these values:
2020-11-12T06:48:59.5801953Z   1U
2020-11-12T06:48:59.5802274Z     Which is: 1
2020-11-12T06:48:59.5802707Z   store_.counter("test.cx_destroy_local_with_active_rq").value()
2020-11-12T06:48:59.5803311Z     Which is: 0
2020-11-12T06:48:59.5803658Z Stack trace:
2020-11-12T06:48:59.5804212Z   0x6ad607: Envoy::Extensions::NetworkFilters::MongoProxy::MongoProxyFilterTest_ConnectionDestroyLocal_Test::TestBody()
2020-11-12T06:48:59.5805303Z   0x1050e78: testing::internal::HandleExceptionsInMethodIfSupported<>()
2020-11-12T06:48:59.5805925Z   0x1050da5: testing::Test::Run()
2020-11-12T06:48:59.5806456Z   0x1051940: testing::TestInfo::Run()
2020-11-12T06:48:59.5806840Z ... Google Test internal frames ...
2020-11-12T06:48:59.5807102Z 
2020-11-12T06:48:59.5807487Z test/extensions/filters/network/mongo_proxy/proxy_test.cc:630: Failure
2020-11-12T06:48:59.5808029Z Expected equality of these values:
2020-11-12T06:48:59.5808351Z   0U
2020-11-12T06:48:59.5808649Z     Which is: 0
2020-11-12T06:48:59.5809211Z   store_.counter("test.cx_destroy_remote_with_active_rq").value()
2020-11-12T06:48:59.5809606Z     Which is: 1
2020-11-12T06:48:59.5809922Z Stack trace:
2020-11-12T06:48:59.5810452Z   0x6ad775: Envoy::Extensions::NetworkFilters::MongoProxy::MongoProxyFilterTest_ConnectionDestroyLocal_Test::TestBody()
2020-11-12T06:48:59.5811100Z   0x1050e78: testing::internal::HandleExceptionsInMethodIfSupported<>()
2020-11-12T06:48:59.5811686Z   0x1050da5: testing::Test::Run()
2020-11-12T06:48:59.5812091Z   0x1051940: testing::TestInfo::Run()
2020-11-12T06:48:59.5812470Z ... Google Test internal frames ...